src/cli/commands/lint.ts

```diff
@@ -150,3 +167,19 @@ async function readInputArguments(name: string, encoding: string) {
   }
   throw new Error(`${name} does not exist`);
 }
+
+function mergeConfig(config: IConfig, flags: any): IConfig {
+  return merge(
```

Comment:
If I follow the code correctly the object is very likely to be 'flat', so we can using `Object.assign` or object spread properties might be a more suitable solution here.

```ts
return {
  ...config,
  ...omitBy<IConfig>(
      {
        encoding: flags.encoding,
        format: flags.format,
        output: flags.output,
        maxResults: flags.maxResults,
        verbose: flags.verbose,
      },
      isNil
  ),
};
```

Reply:
Right, so far it's flat and spread could be better. I just need to be sure that it being flat is what is expected. @philsturgeon

Reply:
Flat sounds good, we can get fancy later